File: sdk/react-native/test_vines.js
import React, { useEffect, useState } from 'react';
import { Text, View, ScrollView } from 'react-native';
import ScyKernel from './ScyKernel'; 
import RNFS from 'react-native-fs'; // Common RN Filesystem lib

const TestVines = () => {
  const [log, setLog] = useState("⏳ Initializing React-Native Parity Test...");

  const runTest = async () => {
    const testKey = "User";
    const testValue = "Amanda";
    const password = "ScyWeb_Global_Secret_2026";
    const dbPath = `${RNFS.DocumentDirectoryPath}/rn_vine.ppm`;

    try {
      // RNFS.writeFile handles the 15-byte header parity
      const header = "P6 4000 4000 255\n".substring(0, 15);
      await RNFS.writeFile(dbPath, header, 'ascii');

      // Note: Most RN file libs don't have ftruncate. 
      // We fill the "soil" with null bytes to reach 48,000,015.
      // In a real app, you'd ship a pre-allocated asset.
      const dummyData = "\0".repeat(1024 * 1024); // 1MB chunk
      for(let i = 0; i < 47; i++) {
          await RNFS.appendFile(dbPath, dummyData, 'ascii');
      }

      // INITIALIZE KERNEL
      const scy = new ScyKernel(password, dbPath);

      // SOW: Put operation (Must use 1600 offset internally)
      await scy.put(testKey, testValue);

      // HARVEST: Get operation
      const result = await scy.get(testKey);

      // CLEANUP & VALIDATION
      if (await RNFS.exists(dbPath)) {
        await RNFS.unlink(dbPath);
      }

      if (result === testValue) {
        setLog(prev => prev + `\n✅ RN KV Parity: SUCCESS\n(Recovered: ${result})`);
      } else {
        setLog(prev => prev + `\n❌ RN KV Parity: FAIL\nExpected: ${testValue}, Got: [${result}]`);
      }

    } catch (err) {
      setLog(prev => prev + `\n❌ RN SDK Error: ${err.message}`);
    }
  };

  useEffect(() => { runTest(); }, []);

  return (
    <ScrollView style={{ padding: 20, backgroundColor: '#1a1a1a' }}>
      <Text style={{ color: '#00ff00', fontFamily: 'monospace' }}>{log}</Text>
    </ScrollView>
  );
};

export default TestVines;

File: sdk/go/test_vines.go
package main

import (
	"fmt"
	"os"
)

func main() {

	const testKey = "User"
	const testValue = "Amanda"
	const password = "ScyWeb_Global_Secret_2026"
	const dbPath = "vines_images/go_vine.ppm"

	_ = os.Mkdir("vines_images", 0755)
	f, err := os.Create(dbPath)
	if err != nil {
		fmt.Printf("❌ IO Error: %v\n", err)
		os.Exit(1)
	}
	
	// Exact 15-byte header: "P6 4000 4000 255\n" truncated to 15
	header := []byte("P6 4000 4000 255\n")
	f.Write(header[:15])
	f.Truncate(48000015)
	f.Close()

	// Ensure NewScyKernel and its methods are visible in the main package
	scy := NewScyKernel(password, dbPath)

	// SOW: Put operation (uses 1600 offset)
	if err := scy.Put(testKey, testValue, password); err != nil {
		fmt.Printf("❌ Put Error: %v\n", err)
		os.Remove(dbPath)
		os.Exit(1)
	}

	// HARVEST: Get operation (uses 1600 offset)
	result, err := scy.Get(testKey, password)
	if err != nil {
		fmt.Printf("❌ Get Error: %v\n", err)
		os.Remove(dbPath)
		os.Exit(1)
	}

	if result == testValue {
		fmt.Printf("✅ Go KV Parity: SUCCESS (Recovered: %s)\n", result)
		scy.DeleteDB(dbPath)
		os.Exit(0)
	} else {
		fmt.Printf("❌ Go KV Parity: FAIL\n")
		fmt.Printf("Expected: %s, Got: [%s]\n", testValue, result)
		scy.DeleteDB(dbPath)
		os.Exit(1)
	}
}